tap_zoom/sync.py

```diff
@@ -9,28 +10,143 @@
 
 LOGGER = singer.get_logger()
 
-def get_bookmark(state, stream_name, default):
-    return state.get('bookmarks', {}).get(stream_name, default)
-
-def write_bookmark(state, stream_name, value):
-    if 'bookmarks' not in state:
-        state['bookmarks'] = {}
-    state['bookmarks'][stream_name] = value
-    singer.write_state(state)
 
 def write_schema(stream):
     schema = stream.schema.to_dict()
     singer.write_schema(stream.tap_stream_id, schema, stream.key_properties)
 
 
+def update_key_bag_for_child(key_bag, parent_endpoint, record):
+    # Constructs the properties needed to build the nested
+    # paths used by the Zoom APIs.
+    # Ex. the list of recordings is fetched at
+    #   /users/{userId}/recordings
+    # We get the list of user records and pass the following
+    # key bag to the recordings endpoint:
+    #   {"userId": <user_id>}
+    updated_key_bag = dict(key_bag)
+
+    if parent_endpoint and 'provides' in parent_endpoint:
+        for dest_key, obj_key in parent_endpoint['provides'].items():
+            updated_key_bag[dest_key] = record[obj_key]
+
+    return updated_key_bag
+
+
+def sync_recordings(client,
+                    catalog,
+                    state,
+                    required_streams,
+                    selected_streams,
+                    stream_name,
+                    endpoint,
+                    key_bag,
+                    parent_endpoint=None,
+                    records=[]):
+    utc_format = '%Y-%m-%d'
+    start_date = singer.get_bookmark(state,
+                                     stream_name,
+                                     'endDate',
+                                     client.start_date)
+    current_date = datetime.utcnow().date()
+    if start_date:
+        start_datetime = singer.utils.strptime_to_utc(start_date)
```

Comment:
this throws a value error since singer expects a datetime-like string instead of a datelike string.

Reply:
@vorpal56 Hmm this fully works for me when I run this code. i am running `singer-python===5.9.0`, which isn't the latest but the method hasn't changed in 5 years and 5.9.0 was released nov 2019.

I verified that it works fine with date-like strings:
```
In [5]: def strptime_to_utc(dtimestr):
   ...:     d_object = dateutil.parser.parse(dtimestr)
   ...:     if d_object.tzinfo is None:
   ...:         return d_object.replace(tzinfo=pytz.UTC)
   ...:     else:
   ...:         return d_object.astimezone(tz=pytz.UTC)
   ...: 

In [6]: strptime_to_utc('2022-11-08')
Out[6]: datetime.datetime(2022, 11, 8, 0, 0, tzinfo=<UTC>)
```

Are you thinking of `singer.strptime`? It's deprecated but it looks like that method only accepts datetime strings:
<img width="824" alt="Screen Shot 2022-11-08 at 5 23 57 PM" src="https://user-images.githubusercontent.com/1455838/200713339-e67db017-59ea-4aaa-9a5b-2ea9e18b97a5.png">